### ISSUE TITLE
Unable to specify multiple properties for task launch

### ISSUE THREAD

**Issue by trisberg** [spring-cloud/spring-cloud-dataflow#1076]
Haven't been able to find any docs for how the `--properties` option for launching a task via the shell is supposed to work.  Tried separating multiple entries with both space and comma, seems neither one works as expected and I can't provide more than one property. Also tried this via the UI and get the second property expression concatenated with the value of the first property. This is the same I get if I separate the properties with a comma using the shell.

**Comment by trisberg** [comment]
Are the deployment properties expected to have "app.timestamp.server.port" style prefix like we do for streams? The behavior seems a bit different for this scenario, although it doesn't seem to work as I would expect with this approach either since the "app.timestamp" prefix isn't stripped off the properties.

**Comment by trisberg** [comment]
We should document the expected way to use this and verify that it works.

**Comment by jvalkeal** [comment]
Yeah, this is wrong. `StreamDeploymentController` extracts props for a specific apps effectively stripping `app.<name>` away from properties. `TaskDeploymentController` doesn't do it and it just passes properties as it. I'm pretty sure this used to work as I've been been using those in yarn. Difficult to say when this got broken.

**Comment by jvalkeal** [comment]
Launching with deploy props and command-line args:
```
task launch --name footask1 --properties "app.timestamp.foo1=bar1,app.timestamp.foo2=bar2" --arguments "foo3=bar3,foo4=bar4"
```
you get command
```
java -jar timestamp-task-1.0.1.RELEASE.jar foo3=bar3 foo4=bar4
```
and these are set to environment(at least with local deployer which should not even pass those into app as those are meant for deployer)
```
app.timestamp.foo2=bar2
app.timestamp.foo1=bar1
```

**Comment by ericbottard** [comment]
There's no doubt this is currently broken. https://github.com/spring-cloud/spring-cloud-dataflow/pull/1047 fixes it

**Comment by jvalkeal** [comment]
Well, what a waste of time that PR was this. We should close duplicate tickets.

**Comment by jvalkeal** [comment]
hmm, I suppose PR I made would fix dataflow 1.x.

**Comment by trisberg** [comment]
We did add a link to the other PR to avoid this, but  yeah, fixing 1.x should be done - maybe merge this PR on the 1.1.x branch and #1047 on master after we create the maintenance branch.

**Comment by ericbottard** [comment]
Please avoid merge nightmare...

**Comment by trisberg** [comment]
Absolutely, this PR should be merged on the 1.1.x branch - after #1058

**Comment by ericbottard** [comment]
Merging two different PRs on the two branches will still lead to conflicts when we'll want to apply changes on those classes that apply to both.
If we really want to fix this in the 1.1 line, we should eg interactively reorder the other PR so that a commit for the "fix" emerges.
Headaches ahead...
